p2p/transport/tcpreuse/reuseport.go

```diff
@@ -11,13 +11,13 @@ import (
 // It default to true.
 const envReuseport = "LIBP2P_TCP_REUSEPORT"
 
-// envReuseportVal stores the value of envReuseport. defaults to true.
-var envReuseportVal = true
+// EnvReuseportVal stores the value of envReuseport. defaults to true.
+var EnvReuseportVal = true
```

Comment:
This doesn't feel good to export as it's only used for tests and is the result of me moving things to avoid dependency cycles (and separate concerns).

Any suggestions on where this reuseport logic lives? Could potentially move the tests or have enough options to thread the testing logic through

Reply:
There's a refactor to be done here, but maybe we do that in a separate PR? This one is already quite big